src/nunit-gui/Model/ResultSummaryReporter.cs

```diff
@@ -0,0 +1,87 @@
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace NUnit.Gui.Model
+{
+    public static class ResultSummaryReporter
+    {
+        public static string WriteSummaryReport(ResultSummary summary)
+        {
+            var sb = new StringBuilder();
+            using (var writer = new StringWriter(sb))
+            {
+                writer.WriteLabelLine("Overall result: ", summary.OverallResult);
+
+                writer.WriteSummaryCount("  Test Count: ", summary.TestCount);
+                writer.WriteSummaryCount(", Passed: ", summary.PassCount);
+                writer.WriteSummaryCount(", Failed: ", summary.FailedCount/*, ColorStyle.Failure*/);
+                writer.WriteSummaryCount(", Warnings: ", summary.WarningCount/*, ColorStyle.Warning*/);
+                writer.WriteSummaryCount(", Inconclusive: ", summary.InconclusiveCount);
+                writer.WriteSummaryCount(", Skipped: ", summary.TotalSkipCount);
+                writer.WriteLine();
+
+                if (summary.FailedCount > 0)
+                {
+                    writer.WriteSummaryCount("    Failed Tests - Failures: ", summary.FailureCount);
+                    writer.WriteSummaryCount(", Errors: ", summary.ErrorCount/*, ColorStyle.Error*/);
+                    writer.WriteSummaryCount(", Invalid: ", summary.InvalidCount);
+                    writer.WriteLine();
+                }
+                if (summary.TotalSkipCount > 0)
+                {
+                    writer.WriteSummaryCount("    Skipped Tests - Ignored: ", summary.IgnoreCount);
+                    writer.WriteSummaryCount(", Explicit: ", summary.ExplicitCount);
+                    writer.WriteSummaryCount(", Other: ", summary.SkipCount);
+                    writer.WriteLine();
+                }
+
+                writer.WriteLabelLine("  Start time: ", summary.StartTime.ToString("u"));
+                writer.WriteLabelLine("    End time: ", summary.EndTime.ToString("u"));
+                writer.WriteLabelLine("    Duration: ", string.Format(NumberFormatInfo.InvariantInfo, "{0:0.000} seconds", summary.Duration));
```

Comment:
This does not align as pretty as in the console (at least using monospaced fonts).